lib/srv/monitor.go

```diff
@@ -424,3 +426,60 @@ func (t *TrackingReadConn) UpdateClientActivity() {
 	defer t.mtx.Unlock()
 	t.lastActive = t.cfg.Clock.Now().UTC()
 }
+
+// GetDisconnectExpiredCertFromIdentity calculates the proper value for DisconnectExpiredCert
+// based on whether a connection is set to disconnect on cert expiry, and whether
+// the cert is a short lived (<1m) one issued for an MFA verified session.
+func GetDisconnectExpiredCertFromIdentity(
+	checker services.AccessChecker,
+	authPref types.AuthPreference,
+	identity *tlsca.Identity,
+) time.Time {
+	// In the case where both disconnect_expired_cert and require_session_mfa are enabled,
+	// the PreviousIdentityExpires value of the certificate will be used, which is the
+	// expiry of the certificate used to issue the short lived MFA verified certificate.
+	//
+	// See https://github.com/gravitational/teleport/issues/18544
+
+	// If the session doesn't need to be disconnected on cert expiry just return the default value.
+	if !checker.AdjustDisconnectExpiredCert(authPref.GetDisconnectExpiredCert()) {
+		return time.Time{}
+	}
+
+	if identity.MFAVerified != "" && !identity.PreviousIdentityExpires.IsZero() {
```

Comment:
Since PreviousIdentityExpires is only set when MFA is required, can't we just use 

```suggestion
	if !identity.PreviousIdentityExpires.IsZero() {
```